platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ImportMigrationManagerImpl.java

```diff
@@ -120,22 +129,21 @@ public ImportMigrationManagerImpl(
       JsonUtils.getMapFrom(metadata, MigrationContextImpl.METADATA_MIGRATABLES)
           .forEach((id, o) -> getContextFor(id).processMetadata(JsonUtils.convertToMap(o)));
     } catch (SecurityException | IOException e) {
-      IOUtils.closeQuietly(zip);
       throw new MigrationException(Messages.IMPORT_FILE_READ_ERROR, exportFile, e);
     } catch (RuntimeException e) {
-      IOUtils.closeQuietly(zip);
       throw e;
     }
   }
 
-  private static ZipFile newZipFileFor(Path exportFile) {
+  private static Map<ZipEntry, byte[]> getImportZipContentsFrom(Path exportFile)
+      throws NoSuchAlgorithmException, NoSuchPaddingException {
     Validate.notNull(exportFile, "invalid null export file");
+    ZipInputStream zis = null;
     try {
-      return new ZipFile(exportFile.toFile());
-    } catch (FileNotFoundException e) {
-      throw new MigrationException(Messages.IMPORT_FILE_MISSING_ERROR, exportFile, e);
-    } catch (SecurityException | IOException e) {
-      throw new MigrationException(Messages.IMPORT_FILE_OPEN_ERROR, exportFile, e);
+      zis = getCipherZipInputStreamFor(exportFile);
```

Comment:
Could we just do a try with resource here instead of the finally block

Reply:
We should not use try-with-resource with input streams. It can throw IOExeptions when closing the streams and we do not care about failing to close once we are done reading.